Jonghakseo/chrome-extension-boilerplate-react-vite
# Add ESLint plugin with Tailwind CSS support and unify project configuration

# Add ESLint plugin with Tailwind CSS support and unify project configuration

## Priority*
- [ ] High: This PR needs to be merged first for other tasks.
- [ ] Middle: This PR should be merged quickly to prevent conflicts due to common changes. (default)

## Purpose of the PR*
This PR adds ESLint with Tailwind CSS support for automatically sorting tailwind classes.

## Changes*
1. Install `eslint-plugin-tailwindcss` as a dev dependency in the root `package.json`.
2. Update root `package.json` with new lint scripts for the entire project.

## How to check the feature
1. Pull the changes and run `pnpm install` to update dependencies.
2. Run `pnpm run lint` to check if ESLint runs successfully across the entire project.
3. Make a small change that violates a Tailwind CSS rule (e.g., use an invalid class name) in a React component.
4. Run `pnpm run lint` again to verify that the Tailwind CSS linting catches the error.
5. Run `pnpm run lint:fix` to check if auto-fixable issues are corrected.
6. Try running `pnpm dev` and `pnpm build` to ensure the new setup doesn't break existing scripts.

## Reference
- [eslint-plugin-tailwindcss documentation](https://github.com/francoismassart/eslint-plugin-tailwindcss)
